src/player.cpp

```diff
@@ -782,35 +781,32 @@ DepotChest* Player::getDepotChest(uint32_t depotId, bool autoCreate)
 {
 	auto it = depotChests.find(depotId);
 	if (it != depotChests.end()) {
-		return it->second;
+		return it->second.get();
```

Comment:
What about returning a smart pointer from this function? Would it be too much to change?

Reply:
Would this make the path in the stack cleaner?

Reply:
No idea. I didn't check.

Reply:
I'll do the test and let you know the result

Reply:
I have done several tests and it seems that as it is now it is very good, since in case of directly handling `DepotChest_ptr` as a return value, it seems to require adding many `.get()` in many parts of the code

The only way would be to create some kind of operator function that does the `.get` transformation directly with the `= ` operator.

Reply:
Should be adressed in another PR I'd say, this is a bug fix so we shouldn't start fiddling around in to much corners